airflow/providers/amazon/aws/hooks/eks.py

```diff
@@ -135,13 +146,12 @@ def create_nodegroup(
         :rtype: Dict
         """
         eks_client = self.conn
+
         # The below tag is mandatory and must have a value of either 'owned' or 'shared'
         # A value of 'owned' denotes that the subnets are exclusive to the nodegroup.
         # The 'shared' value allows more than one resource to use the subnet.
-        tags = {'kubernetes.io/cluster/' + clusterName: 'owned'}
-        if "tags" in kwargs:
-            tags = {**tags, **kwargs["tags"]}
-            kwargs.pop("tags")
+        tags = kwargs.pop("tags", {})
+        tags[f'kubernetes.io/cluster/{clusterName}'] = 'owned'
```

Comment:
I thought of that when I was looking at @uranusjr 's comment above but wanted to test it to see.  Would it have preserved the user's setting the way it was before?

Reply:
If “before” means the code currently in `main`, yes it would, because `{**tags, **kwargs["tags"]}` means that if any key in `kwargs["tags"]` would override the value under the same key in `tags`.

```python
>>> a = {"x": 1}
>>> b = {"x": 2}
>>> {**a, **b}
{"x": 2}
```

So the backward compatible implementation would be something like

```python
resolved_tags = tags or {}
cluster_tag_key = f'kubernetes.io/cluster/{clusterName}'
if cluster_tag_key not in resolved_tags:
    resolved_tags[cluster_tag_key] = 'owned'
```

or `setdefault` might work as well.

Reply:
in effect the current code has a default of `owned` and i think that should be preserved without good reason for changing

Reply:
Yes, I couldn't remember offhand if the previous/existing implementation `{**tags, **kwargs["tags"]}` overwrote the default with the user's or not.  I was going to confirm how it worked before I said anything, but @dstandish caught it first.

Reply:
@uranusjr - Is there a particular reason why you are making the tags parameter keyword-only in your solution here?

Reply:
The parameter was retrieved with `kwargs.pop()`, so it can only be passed as a keyword argument to begin with. I don’t have particular reasons to make it keyword-only, but since it already is, the keyword-only argument syntax makes it more explicit.

Reply:
I guess that makes sense.  I've added that in https://github.com/apache/airflow/pull/20819/commits/cc12d26eb4fea209ae5c616feaffa94c95cf8a62, just wanted to make sure I understood why.  Thanks